### PR TITLE
[MM-28238] Enable deployment of load-test agents into custom VPC

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR enables load-test agent instances to be associated with a provided VPC by adding two optional config settings `ClusterVpcID` and `ClusterSubnetID`.

#### Ticket

https://mattermost.atlassian.net/browse/MM-28238

### CODE REVIEW

File: deployment/terraform/create.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ const (
 	defaultLoadTestDownloadURL = "https://github.com/mattermost/mattermost-load-test-ng/releases/download/v1.0.0/mattermost-load-test-ng-v1.0.0-linux-amd64.tar.gz"
 	filePrefix                 = "file://"
 	minSupportedVersion        = 0.12
-	maxSupportedVersion        = 0.12
+	maxSupportedVersion        = 0.13
```
Comment (streamer45) [inline deployment/terraform/create.go:31]: Bumping the max supported version since it appears to be working fine.

File: docs/deployer_config.md [hunk 2]
```diff
@@ -6,6 +6,18 @@
 
 The name of the cluster. This will be prefixed to all resources in AWS that are built with the configuration.
 
+## ClusterVpcID
+
+*string*
+
+The ID of the VPC associated to the resources.
```
Comment (isacikgoz) [inline docs/deployer_config.md:13]: Maybe add a note that this is meant to be used in pre-deployed environments?

**Review by isacikgoz — APPROVED** [review]
Awesome! Just a thing below:

File: deployment/terraform/assets/cluster.tf [hunk 3]
```diff
@@ -305,10 +305,12 @@ resource "aws_instance" "loadtest_agent" {
...
 
-  ami           = "ami-0ac80df6eff0e70b5"
-  instance_type = var.agent_instance_type
-  key_name      = aws_key_pair.key.id
-  count         = var.agent_instance_count
+  ami                         = "ami-0ac80df6eff0e70b5"
+  instance_type               = var.agent_instance_type
+  key_name                    = aws_key_pair.key.id
+  count                       = var.agent_instance_count
+  subnet_id                   = var.cluster_subnet_id
```
Comment (stylianosrigas) [inline deployment/terraform/assets/cluster.tf:312]: @streamer45 Just a question on that. Noticed that we are providing the subnet_id only for the loadtest agent. Don't we need that for the other instances as well?

**Review by stylianosrigas — APPROVED** [review]
LGTM, just one question.

File: deployment/terraform/assets/cluster.tf (hunk 3 above)
Comment (streamer45) [reply deployment/terraform/assets/cluster.tf:312]: If we wanted to support deploying everything under the same subnet/VPC then yes but my understanding was that you only require the agents, everything else (MM app cluster and Prometheus/Grafana) is deployed separately so here I am just covering this special case. If that's not sufficient we can extend this to all other instances.

Comment (stylianosrigas) [reply deployment/terraform/assets/cluster.tf:312]: Sorry didn't get that metrics and app was referring to the Mattermost app and the monitoring part. Makes total sense, thanks for making it clear 👍

**Pull request merged**